File: STRIVER/video/vid-7/count-digits.cpp
#include <bits/stdc++.h>
using namespace std;


// time complexity  = log10(n)

int main()
{
    // cout << "Hola!!!"<<endl;
    int N = 2223232;
    int n = N;
    int counts = 0;

    while (n != 0)
    {
        int temp = n % 10;
        if (temp != 0 && N % temp == 0)
        {
            counts++;
        }
        n = n / 10;
    }
    cout << counts << endl;
    return counts;
}

File: STRIVER/video/vid-7 (Maths)/count-digits.py
#  time complexity  = log10(n)

def evenlyDivides(N):
    # code here
    n = N
    divs = 0
    
    while n!=0 :
        temp = n%10
        if(temp != 0 and N%temp == 0):
            divs+=1
        n = n//10
    return divs

print(evenlyDivides(232))